internal/cli/cli.go

```diff
@@ -48,40 +46,44 @@ func NewCli(runner Runner, fs afero.Fs) *Cli {
 	}
 }
 
-func (c *Cli) run(cmd *exec.Cmd) (exit int, reterr error) {
-	cmd.Stdin = os.Stdin
-	cmd.Stderr = os.Stderr
-	stdout, err := cmd.StdoutPipe()
+func (c *Cli) run(cmd *exec.Cmd) (int, error) {
+	// force line buffering for stdout
+	// otherwise it will be fully buffered because our stdout is not a tty
+	path, err := exec.LookPath("stdbuf")
 	if err != nil {
-		panic(err)
-	}
-	if err := c.runner.Start(cmd); err != nil {
-		panic(err)
+		return 1, err
 	}
-	reader := bufio.NewReader(stdout)
-	for {
-		s, err := reader.ReadString('\n')
-		if err != nil {
-			break
-		}
-		if reterr = findCommonError(s); reterr != nil {
-			break
-		}
-		fmt.Print(s)
-	}
-	if err := c.runner.Wait(cmd); err != nil {
+	cmd.Path = path
+	cmd.Args = append([]string{"stdbuf", "-oL"}, cmd.Args...)
+
+	cmd.Stdin = os.Stdin
+
+	// capture stdout and stderr
+	var stdout, stderr cappedBuffer
+	cmd.Stdout = io.MultiWriter(os.Stdout, &stdout)
+	cmd.Stderr = io.MultiWriter(os.Stderr, &stderr)
```

Comment:
Using MultiWriter, messy error messages will still be printed to users console. With StdoutPipe, we can replace them completely.